x/auth/tx/service_test.go

```diff
@@ -565,6 +567,110 @@ func (s *IntegrationTestSuite) TestSimMultiSigTx() {
 	s.Require().Greater(res.GasInfo.GasUsed, uint64(0))
 }
 
+// Tests that all msg events included in an authz MsgExec tx
+// Ref: https://github.com/cosmos/cosmos-sdk/issues/9501
+func (s *IntegrationTestSuite) TestSim_MsgExecEvents() {
+	val1 := s.network.Validators[0]
+	val1.ClientCtx.Offline = true
+
+	// create a second account and fund it for grantee usage
+	kr := val1.ClientCtx.Keyring
+	acc1, _, err := kr.NewMnemonic("granteeAccount", keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	s.Require().NoError(err)
+	out, err := bankcli.MsgSendExec(
+		val1.ClientCtx,
+		val1.Address,
+		acc1.GetAddress(),
+		sdk.NewCoins(
+			sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(100)),
+		),
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+		fmt.Sprintf("--gas=%d", flags.DefaultGasLimit),
+		fmt.Sprintf("--%s=foobar", flags.FlagNote),
+	)
```

Comment:
why do we need this call? 
I think this call should fail because we don't have any authorization for `acc1` created at this stage.

Reply:
it's not an auth tx - its a bank tx sending coins from val to grantee account. have to fund the grantee account otherwise we cannot use it in the next tx's in this test

Reply:
oh, this is very obscure way to fund account. If you have access to the bank keeper then you can do something like:
```
require := suite.Require()
balances := sdk.NewCoins(newFooCoin(100), newBarCoin(50))
addr1 := sdk.AccAddress("addr1_______________")
app.AccountKeeper.SetAccount(ctx, acc1)
require.NoError(simapp.FundAccount(app.BankKeeper, ctx, addr1, balances))
```